### ISSUE TITLE
Partial matches on "onlyAutocomplete"

### ISSUE THREAD

**Issue by theguy147** [agustinl/svelte-tags-input#49]
Currently when adding a autocomplete list to `svelte-tags-input` and setting `onlyAutocomplete` partial matches of the autocomplete list are still allowed.

Is this a bug or is this the expected behavior? This feels wrong to me (IMHO partial matches should be disallowed)!

### Example ###

This is a minimal example showing this behavior (e.g. insert "f" and press "Enter". Now "f" was accepted as a Tag and not "foo" as expected). The example can be tried out in this [REPL](https://svelte.dev/repl/c3b6446dafe844f6b203645b2f91f515?version=3.42.4).

```svelte
<script>
	import Tags from "svelte-tags-input";
	let autocomplete = ["foo", "bar"];
</script>

<Tags autoComplete={autocomplete} onlyAutocomplete />
```

Edit: I found a related issue: #8. So if I understand correctly it was working differently at some point (at least with TAB key)?

**Comment by theguy147** [comment]
it's not the exact solution to my issue but it works in my case and therefore i wanted to share it in case other people are intersted:

I added an options to change the autoComplete key(s) from "TAB" to something else and setting it to the same key(s) as `addKeys` then always autocompletes the partial match to the next autocomplete match in the list. 

This is my patch: 

```diff
diff --git a/src/Tags.svelte b/src/Tags.svelte
index c9fbc39..7e0526b 100644
--- a/src/Tags.svelte
+++ b/src/Tags.svelte
@@ -20,6 +20,7 @@ export let allowDrop;
 export let splitWith;
 export let autoComplete;
 export let autoCompleteKey;
+export let autoCompleteKeys;
 export let name;
 export let id;
 export let allowBlur;
@@ -40,6 +41,7 @@ $: allowDrop = allowDrop || false;
 $: splitWith = splitWith || ",";
 $: autoComplete = autoComplete || false;
 $: autoCompleteKey = autoCompleteKey || false;
+$: autoCompleteKeys = autoCompleteKeys || [9];
 $: name = name || "svelte-tags-input";
 $: id = id || uniqueID();
 $: allowBlur = allowBlur || false;
@@ -62,19 +64,16 @@ function setTag(input) {
             if (key === input.keyCode) {
                 
                 if (currentTag) input.preventDefault();
-                                
-                switch (input.keyCode) {
-                case 9:
-                    // TAB add first element on the autoComplete list
-                    if (autoComplete && document.getElementById(matchsID)) {                        
+
+                if (autoCompleteKeys.includes(input.keyCode)) {
+                     // autoCompleteKeys add first element on the autoComplete list
+                    if (autoComplete && document.getElementById(matchsID)) {
                         addTag(document.getElementById(matchsID).querySelectorAll("li")[0].textContent);
                     } else {
                         addTag(currentTag);
-                    }                    
-                    break;
-                default:
+                    }
+                } else {
                     addTag(currentTag);
-                    break;
                 }
             }
         });
@@ -501,4 +500,4 @@ function uniqueID() {
     white-space: nowrap;
     border: 0;
 }
-</style>
\ No newline at end of file
+</style>
```

In case this is interesting for this repo obviously the documentation would have to be adjusted to include this new option as well.

**Comment by agustinl** [comment]
Hi @theguy147 
Sorry for the late response. 
You're right, this is a bug. I fixed with something like your answer #50 . 

```diff
@@ -54,7 +54,6 @@ $: matchsID = id + "_matchs";
let storePlaceholder = placeholder;
function setTag(input) {
        
    const currentTag = input.target.value;
    
    if (addKeys) {
@@ -63,7 +62,7 @@ function setTag(input) {
                
                if (currentTag) input.preventDefault();
                                
-                switch (input.keyCode) {
-                /* switch (input.keyCode) {
-                case 9:
-                    // TAB add first element on the autoComplete list
-                    if (autoComplete && document.getElementById(matchsID)) {                        
-@@ -75,6 +74,11 @@ function setTag(input) {
-                default:
-                    addTag(currentTag);
-                    break;
-                } */
+                if (autoComplete && document.getElementById(matchsID)) {
+                    addTag(document.getElementById(matchsID).querySelectorAll("li")[0].textContent);
+                } else {
+                    addTag(currentTag);
+                }
            }
        });
```

I checked and all scenarios works well. Could you check from your side? I will push this to prod the next friday.

Thanks!

**Comment by theguy147** [comment]
@agustinl 
Yes, this works fine for me! Thanks.
